### PR TITLE
Feature/maroon 394 coord voltmeter

### LINKED ISSUE #394: [Coord System] Voltmeter
The implementation of an reusable voltmeter

### PR DESCRIPTION
**Pull request by dpdsw**
closes #394

### CODE REVIEW

File: unity/Assets/Maroon/reusableGui/Experiment/Resources/Tools/VoltmeterUI.prefab [hunk 1]
```diff
@@ -0,0 +1,8559 @@
+%YAML 1.1
```
Comment (michaelholly) [inline unity/Assets/Maroon/reusableGui/Experiment/Resources/Tools/VoltmeterUI.prefab]: Maybe rename VoltmeterUI and RulerUI to Voltmeter and Ruler

File: unity/Assets/Maroon/reusablePrefabs/NewVoltmeter/Example/VoltmeterExample.unity [hunk 2]
```diff
@@ -0,0 +1,13785 @@
+%YAML 1.1
```
Comment (michaelholly) [inline unity/Assets/Maroon/reusablePrefabs/NewVoltmeter/Example/VoltmeterExample.unity]: ArgumentOutOfRangeException in the play mode

File: unity/Assets/Maroon/reusablePrefabs/NewVoltmeter/Scripts/VoltmeterCharge.cs [hunk 3]
```diff
@@ -0,0 +1,81 @@
+﻿using Maroon.GlobalEntities;
+using Maroon.Physics;
+using UnityEngine;
+
+namespace Maroon.Tools.Voltmeter
+{
+    public class VoltmeterCharge : MonoBehaviour, IGenerateE
```
Comment (michaelholly) [inline unity/Assets/Maroon/reusablePrefabs/NewVoltmeter/Scripts/VoltmeterCharge.cs:7]: why another charge?

Charge in Assets\Maroon\reusableScripts\Physics\Electromagnetism
Charge in Assets\Maroon\scenes\experiments\Capacitor\Scripts
CoulombChargeBehaviour in Assets\Maroon\scenes\experiments\CoulombsLaw\Scripts

Please reuse the Charge in Assets\Maroon\reusableScripts\Physics\Electromagnetism for your purpose

File: unity/Assets/Maroon/reusablePrefabs/NewVoltmeter/Scripts/VoltmeterPinPoint.cs [hunk 4]
```diff
@@ -0,0 +1,32 @@
+﻿using System;
+using UnityEngine;
+using UnityEngine.Events;
+
+namespace Maroon.Tools.Voltmeter
+{
+    [Serializable] public class VoltmeterEvent : UnityEvent<string> { }
+    public class VoltmeterPinPoint : MonoBehaviour, IResetObject
```
Comment (michaelholly) [inline unity/Assets/Maroon/reusablePrefabs/NewVoltmeter/Scripts/VoltmeterPinPoint.cs]: why does this class not inherit from the Pin class?

**Review by michaelholly — CHANGES_REQUESTED** [review]

File: unity/Assets/Maroon/reusableGui/Experiment/Resources/Tools/VoltmeterUI.prefab (hunk 1 above)
Comment (dpdsw) [reply unity/Assets/Maroon/reusableGui/Experiment/Resources/Tools/VoltmeterUI.prefab]: Done

File: unity/Assets/Maroon/reusablePrefabs/NewVoltmeter/Example/VoltmeterExample.unity (hunk 2 above)
Comment (dpdsw) [reply unity/Assets/Maroon/reusablePrefabs/NewVoltmeter/Example/VoltmeterExample.unity]: SceneManager expects '.' in the scene name. Renamed Example scenes

File: unity/Assets/Maroon/reusablePrefabs/NewVoltmeter/Scripts/VoltmeterPinPoint.cs (hunk 4 above)
Comment (dpdsw) [reply unity/Assets/Maroon/reusablePrefabs/NewVoltmeter/Scripts/VoltmeterPinPoint.cs]: It does now

File: unity/Assets/Maroon/reusablePrefabs/NewVoltmeter/Scripts/VoltmeterCharge.cs (hunk 3 above)
Comment (dpdsw) [reply unity/Assets/Maroon/reusablePrefabs/NewVoltmeter/Scripts/VoltmeterCharge.cs:7]: Done

**Comment by michaelholly** [comment]
Bug in Voltmeter UI: if the user drags the voltmeter pins from the UI into the scene, the position in the UI view is not updated.
The fix would be to call the UpdateDisplay methods in the OnEnabled event in the PC_Drag Handler for VoltmeterPos and VoltmeterNeg
 
image

**Review by michaelholly — APPROVED** [review]

**Pull request merged**